nomad/structs/volumes.go

```diff
@@ -0,0 +1,188 @@
+package structs
+
+import (
+	"github.com/mitchellh/copystructure"
+	"github.com/mitchellh/mapstructure"
+)
+
+const (
+	VolumeTypeHost = "host"
+)
+
+// ClientHostVolumeConfig is used to configure access to host paths on a Nomad Client
+type ClientHostVolumeConfig struct {
+	Name     string `hcl:",key"`
+	Source   string `hcl:"source"`
+	ReadOnly bool   `hcl:"read_only"`
+	Hidden   bool   `hcl:"hidden"`
+}
+
+func (p *ClientHostVolumeConfig) Copy() *ClientHostVolumeConfig {
+	if p == nil {
+		return nil
+	}
+
+	c := new(ClientHostVolumeConfig)
+	*c = *p
+	return c
+}
+
+func CopyMapStringClientHostVolumeConfig(m map[string]*ClientHostVolumeConfig) map[string]*ClientHostVolumeConfig {
+	if m == nil {
+		return nil
+	}
+
+	nm := make(map[string]*ClientHostVolumeConfig, len(m))
+	for k, v := range m {
+		nm[k] = v.Copy()
+	}
+
+	return nm
+}
+
+func HostVolumeSliceMerge(a, b []*ClientHostVolumeConfig) []*ClientHostVolumeConfig {
+	n := make([]*ClientHostVolumeConfig, len(a))
```

Comment:
Assuming most of the time, these are unique, we should initialize the slice capacity to the sum of lengths?

```suggestion
	n := make([]*ClientHostVolumeConfig, len(a)+len(b))
```

Reply:
because we end up with nil initialized values in the slice (https://play.golang.org/p/xBGkp3P1fur)

Reply:
Ah, yes - then you pass it in the capacity area where you don't have this problem https://play.golang.org/p/G_pIwXwbhr4 .

```suggestion
	n := make([]*ClientHostVolumeConfig, 0, len(a)+len(b))
```